owncloudApp/src/main/java/com/owncloud/android/utils/NotificationUtils.kt

```diff
@@ -0,0 +1,106 @@
+/**
+ * ownCloud Android client application
+ *
+ *
+ * Copyright (C) 2020 ownCloud GmbH.
+ *
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.utils
+
+import android.accounts.Account
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Process
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import com.owncloud.android.R
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.ui.activity.ConflictsResolveActivity
+import java.util.Random
+
+object NotificationUtils {
+
+    @JvmStatic
+    fun newNotificationBuilder(context: Context, channelId: String): NotificationCompat.Builder {
+        return NotificationCompat.Builder(context, channelId).apply {
+            color = ContextCompat.getColor(context, R.color.primary)
+            setSmallIcon(R.drawable.notification_icon)
+        }
+    }
+
+    @JvmStatic
+    fun cancelWithDelay(
+        notificationManager: NotificationManager,
+        notificationId: Int,
+        delayInMillis: Long
+    ) {
+        val thread = HandlerThread(
+            "NotificationDelayerThread_" + Random(System.currentTimeMillis()).nextInt(),
+            Process.THREAD_PRIORITY_BACKGROUND
+        )
+        thread.start()
+        val handler = Handler(thread.looper)
+        handler.postDelayed({
+            notificationManager.cancel(notificationId)
+            (Thread.currentThread() as HandlerThread).looper.quit()
+        }, delayInMillis)
+    }
+
+    /**
+     * Show a notification with file conflict information, that will open a dialog to solve it when tapping it
+     *
+     * @param fileInConflict file in conflict
+     * @param account        account which the file in conflict belongs to
+     */
+    @JvmStatic
+    fun notifyConflict(fileInConflict: OCFile, account: Account?, context: Context) {
```

Comment:
Please reduce the size of this method by braking it apart into logical smaller pieces.